api/src/main/java/io/keploy/service/GrpcService.java

```diff
@@ -155,6 +164,75 @@ public static void put(Service.TestCaseReq testCaseReq) {
         }
     }
 
+    /**
+     * Filter.  This method is used to filter the test cases based on the regex provided in the configuration
+     *
+     * @param testCaseReq the test case req object which contains all the information about the test case
+     * @return the boolean  true if the test case is valid to be recorded, false otherwise
+     */
+    public static boolean isValidTestCaseRequest(Service.TestCaseReq testCaseReq) {
+        Filter filter = k.getCfg().getApp().getFilter();
+        Pattern pattern = Pattern.compile(filter.getAcceptUrlRegex());
+
+        // if headerRegex is not null and header is not valid, return
+        // if urlRegex is not null and url is not valid, return
+        boolean inValidCondition = (filter.getHeaderRegex() != null && !isValidHeader(testCaseReq))
+                || filter.getAcceptUrlRegex() != "" && pattern.matcher(testCaseReq.getURI()).matches()
+                || filter.getRejectUrlRegex() == null && isExcludedHeader(testCaseReq);
+        return !inValidCondition;
+    }
+
+    /**
+     * Is valid header boolean. This method is used to check if the header is valid or not.
+     *
+     * @param testCaseReq the test case req object which contains all the information about the test case.
+     * @return the boolean  true if header is valid, false otherwise.
+     */
+    public static boolean isValidHeader(Service.TestCaseReq testCaseReq) {
+        Filter filter = k.getCfg().getApp().getFilter();
+        // get header map
+        Map<String, Service.StrArr> testCaseHeaderMap = testCaseReq.getHttpReq().getHeaderMap();
+        boolean validHeader = false;
+        // loop over filter header regex
+        for (String value : filter.getHeaderRegex()) {
+            Pattern pattern = Pattern.compile(value);
+            // loop over header map
+            for (Map.Entry<String, Service.StrArr> entry : testCaseHeaderMap.entrySet()) {
+                // if header key matches with regex, return true
+                String key = entry.getKey();
+                if (pattern.matcher(key).matches()) {
+                    validHeader = true;
+                    break;
+                }
+            }
+        }
+        return validHeader;
+    }
+
+    /**
+     * Is excluded header boolean. This method is used to check if the request url should be excluded or not.
+     *
+     * @param testCaseReq the test case req the test case req object which contains all the information about the test case
+     * @return the boolean  true if header url should be excluded, false otherwise.
+     */
+    public static boolean isExcludedHeader(Service.TestCaseReq testCaseReq) {
```

Comment:
The method name should be something like `isExcludedURL` since you are checking for URL not for header.